ldms/src/core/ldms_xprt.c

```diff
@@ -2510,30 +2569,32 @@ static void handle_rendezvous_push(zap_ep_t zep, zap_event_t ev,
 		return;
 	}
 
-	/* See if we already have a push RBD for this set and
-	 * transport.
-	 */
-	push_rbd = ldms_lookup_rbd(x, set);
-	if (push_rbd) {
+	/* See if we already have a push RBD for this set and transport. */
+	pthread_mutex_lock(&x->lock);
+	push_rbd = __rbd_by_set_first(x, set);
+	if (!push_rbd)
+		goto create_push_rbd;
+	for (; push_rbd; push_rbd = __rbd_by_set_next(push_rbd)) {
 		if (push_rbd->push_flags & LDMS_RBD_F_PUSH) {
-			ref_get(&push_rbd->set->ref, "rendezvous_push");
 			/* Update the push flags, but otherwise, do nothing */
 			goto out;
 		}
 	}
 
+create_push_rbd:
 	/* We will be the target of RDMA_WRITE */
```

Comment:
This comment isn't true... We should remove it.

Reply:
Done.